### PR TITLE
feat: support rabbitmq headers

### PR DESCRIPTION
**Pull request by DaAlbrecht**
## Purpose

This pull request adds the ability to include custom headers in RabbitMQ message properties. It introduces support for the headers field in BasicProperties and updates the Java code to map these headers during message publishing

## Examples

```bal
import ballerinax/rabbitmq;

rabbitmq:ConnectionConfiguration config = {
    username: "guest",
    password: "guest"
};
rabbitmq:Client rabbitmqClient = check new ("localhost", 5672, config);

public function main() returns error? {
    map<anydata> headers = {"header1": "value1", "header2": "value2"};
    rabbitmq:BasicProperties basicProperties = {headers: headers};
    rabbitmq:AnydataMessage anydataMessage = {routingKey: "test", content: "hello world", properties: basicProperties};
    check rabbitmqClient->publishMessage(anydataMessage);
}
```

## Checklist
- [ ] Linked to an issue
- [ ] Updated the changelog
- [x] Added tests
- [ ] Updated the spec
- [x] Checked native-image compatibility

### CODE REVIEW

**Comment by aashikam** [comment]
Hi, Could you add a test for this?

**Comment by DaAlbrecht** [comment]
Hi @aashikam 

The tests didn't pass because I haven't added support for consuming messages and the corresponding headers. Fixed this now and added a new test.

**Comment by tharindu-nw** [comment]
Some compiler plugin tests seem to be failing still

**Comment by aashikam** [comment]
Screenshot 2024-09-27 at 21 28 29

I inspected the error. This comes for the following files but they have correctly used the `rabbitmq:AnydataMessage` record. 
Screenshot 2024-09-27 at 21 31 49

And the PR does not include any compiler plugin related changes.

**Comment by DaAlbrecht** [comment]
Thank you for the fix! @aashikam  merged your pr

**Review by aashikam — APPROVED** [review]
LGTM

File: ballerina/rabbitmq_commons.bal [hunk 1]
```diff
@@ -43,11 +43,13 @@ public const TOPIC_EXCHANGE = "topic";
 # + contentType - The content type of the message
 # + contentEncoding - The content encoding of the message
 # + correlationId - The client-specific ID that can be used to mark or identify messages between clients
+# + headers - A map of additional arbitrary headers to be included in the message.
```
Comment (aashikam) [inline ballerina/rabbitmq_commons.bal:46]:
```suggestion
# + headers - A map of additional arbitrary headers to be included in the message
```

**Review by aashikam — APPROVED** [review]

**Pull request merged**